include/openthread/platform/radio.h

```diff
@@ -159,6 +159,29 @@ struct otExtAddress
  */
 typedef struct otExtAddress otExtAddress;
 
+/**
+ * Converts an array representing an IEEE 802.15.4 Extended Address into a
+ * 64-bit value.
+ *
+ * This is necessary as the otExtAddress array is not necessary aligned for
+ * access as a 64-bit value.
+ */
+static inline uint64_t otExtAddressLoadUint64(const otExtAddress *address)
```

Comment:
@srickardti, is this necessary?

In https://github.com/openthread/openthread/pull/4323#issuecomment-553178110 you had commented:

> I'm not really clear on why this is necessary. The pointer in question was used only to cast between the otExtAddress type and a uint64_t. The M4F or M3 can handle unaligned accesses, and the M0 radio controller is protected by the alignment of the structs it is sent.

In https://github.com/openthread/openthread/pull/4323#discussion_r345513880, you had commented:

> This could be done with a cast and assignment to a uint64_t. I think GCC is smart enough to optimize out the function call, so that is probably fine.

Thoughts?

Reply:
I'm a little unclear what the point of this function is, and the doxygen comment is as well. This function is not aligning access, it is loading an array of little-endian order bytes into a `uint64_t`. Which is then returned to the caller. Alignment in that case is actually enforced by the caller.

Since both the CC1352 and CC2652 are little endian it is equivalent to the cast and de-reference hack that was there before. And GCC seems to be intelligent enough to figure that out: https://godbolt.org/z/qGXJc6. My only concern would be the TI and IAR compilers being as clever.